### PR TITLE
Replace unnecessary use of `aiida_profile_clean`

### LINKED ISSUE #5770: Remove the unnecessary use of `aiida_profile_clean`
A lot of tests use `aiida_profile_clean` even though they don't really require a clean profile. Since the fixture clears the database, which has a certain costs, it is best to avoid it if not needed.

### PR DESCRIPTION
**Pull request by sphuber**
Fixes #5770

### CODE REVIEW

**Comment by sphuber** [comment]
Anyone for this one? @unkcpz @mbercx @ramirezfranciscof ?

File: tests/cmdline/commands/test_group_ls.py [hunk 1]
```diff
@@ -19,8 +19,9 @@
 
 
 @pytest.fixture
-def setup_groups(aiida_profile_clean):
+def setup_groups():
```
Comment (unkcpz) [inline tests/cmdline/commands/test_group_ls.py:22]: Here it seems more clear to use `aiida_profile_clean`.

**Review by unkcpz — COMMENTED** [review]
Not finished yet. I'll continue after the group meeting. 
A quick question, after this PR, how long it will save to run all tests?
If I understand correctly, there is only one profile during a pytest?

**Comment by sphuber** [comment]
> A quick question, after this PR, how long it will save to run all tests?

Unfortunately the run time of the tests on Github Actions are not very stable so it is difficult to say with certainty what the exact speed up is. But the tests of this PR ran in [13m 37s](https://github.com/aiidateam/aiida-core/actions/runs/3628227287/jobs/6119061601) and [13m 27s](https://github.com/aiidateam/aiida-core/actions/runs/3628227287/jobs/6119061733). For comparison, the latest run on the `main` branch ran in [15m 5s](https://github.com/aiidateam/aiida-core/actions/runs/3629167575/jobs/6121047334) and [18m 3s](https://github.com/aiidateam/aiida-core/actions/runs/3629167575/jobs/6121047451). So there seems to be a significant speedup as I would expect, but it can vary.

> If I understand correctly, there is only one profile during a pytest?

Mostly yes. Certain tests may create an additional profile on the fly for just that test.

File: tests/tools/archive/orm/test_codes.py [hunk 2]
```diff
@@ -14,7 +14,7 @@
 from tests.tools.archive.utils import get_all_node_links
 
 
-def test_that_solo_code_is_exported_correctly(tmp_path, aiida_profile_clean, aiida_localhost):
+def test_that_solo_code_is_exported_correctly(aiida_profile, tmp_path, aiida_localhost):
     """
     This test checks that when a calculation is exported then the
     corresponding code is also exported.
```
[the remark this answers is not in the document]
Comment (unkcpz) [reply tests/tools/archive/orm/test_codes.py:20]: Also below line 30 there is a code_label.

**Review by unkcpz — CHANGES_REQUESTED** [review]
@sphuber Thanks! Just some minor requests.

File: tests/tools/archive/orm/test_codes.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (sphuber) [reply tests/tools/archive/orm/test_codes.py:22]: Not sure if that is what you meant, but I have used uuids for the code labels

File: tests/orm/test_querybuilder.py [hunk 3]
```diff
@@ -23,9 +24,9 @@
 from aiida.orm.utils.links import LinkQuadruple
 
 
-@pytest.mark.usefixtures('aiida_profile_clean')
```
[the remark this answers is not in the document]
Comment (unkcpz) [reply tests/orm/test_querybuilder.py:26]: I see, makes sense.

File: tests/tools/archive/orm/test_codes.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (unkcpz) [reply tests/tools/archive/orm/test_codes.py:22]: Yes, that's what I mean. Thanks!

**Review by unkcpz — APPROVED** [review]
Looks all good to me. Cheers!

**Pull request merged**